MachUpskillingFY17/JabbR-Core
# Justine/o auth

To Test:

First set up your user secrets with the Google and Facebook API keys **(email me for the keys)**
 - In the JabbR_Core/src directory run `dotnet user-secrets set Authentication:Facebook:AppID {ID}`
 - Repeat for Facebook App Secret and Google App ID/ Secret

Then set up your environment to use SSL
 - Your launchSettings.json should have already been changed in this PR
    - Verify by right clicking the JabbR-Core project and selecting Properties from the drop down
    - Enable SSL should be selected and your app URL should match the URL listed next to Enable SSL
 - You will need to add the SSL certificate to localhost
    - Run your app WITHOUT debug mode (ctrl + F5) and a popup asking you to add the SSL Certificate for localhost should pop up, select Ok

Now you are ready to test
 - Run the app and you will be on a blank white page
 - Add `/account/login` to navigate to the login page
 - Select "Google" as your login method
    - You should be redirected to the Google sign in page, sign in using your personal Google credentials 
    - You should now be redirected to a JabbR confirmation page where you enter a username and confirm your email address
    - Now you should be logged in as that user and able to create rooms and message in them
    - If you log out, close the browser tab and run the program again you should be able to log in using your Google account again and redirect directly to JabbR without hitting the JabbR registration confirmation page
 - Select "Facebook" as your login method and repeat above steps
